cruise-control-metrics-reporter/src/main/java/com/linkedin/kafka/cruisecontrol/metricsreporter/CruiseControlMetricsReporter.java

```diff
@@ -113,8 +113,19 @@ public void configure(Map<String, ?> configs) {
 
     //Add BootstrapServers if not set
     if (!producerProps.containsKey(CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG)) {
-      Object port = configs.get("port");
-      String bootstrapServers = "localhost:" + (port == null ? "9092" : String.valueOf(port));
+      String listeners = configs.get("listeners") == null ? null : String.valueOf(configs.get("listeners"));
+      String bootstrapServers = null;
+      if (listeners != null && listeners.length() != 0) {
+        String firstListener = listeners.split("\\s*,\\s*")[0];
+        String[] allListeners = firstListener.split(":");
+        if (allListeners[1].length() == 2) {
+          bootstrapServers = "0.0.0.0:" + allListeners[2];
```

Comment:
https://github.com/linkedin/cruise-control/pull/1246#discussion_r448047726 does not seem to be addressed.
Can we avoid using hardcoded values such as `"listeners"` or `"0.0.0.0:"`?

* For config names, we can use the original source of the config names, such as (1) `KafkaConfig.ListenersProp()` and (2) `KafkaConfig.PortProp()`.
* For default values, we can use static variables -- e.g.
```
  protected static final String DEFAULT_BOOTSTRAP_SERVERS_PORT = "9092";
  protected static final String DEFAULT_BOOTSTRAP_SERVERS_HOST = "localhost";
```